driver/bpf/fillers.h

```diff
@@ -2248,6 +2256,29 @@ static __always_inline bool get_exe_writable(struct task_struct *task)
 	return false;
 }
 
+static __always_inline bool get_exe_upper_layer(struct inode *inode)
+{
+	struct super_block *sb = _READ(inode->i_sb);
+	unsigned long sb_magic = _READ(sb->s_magic);
+
+	if(sb_magic == PPM_OVERLAYFS_SUPER_MAGIC)
+	{
+		struct dentry *upper_dentry;
+		struct inode *lower_inode;
+		char *vfs_inode = (char *)inode;
+		
+		bpf_probe_read(&upper_dentry, sizeof(upper_dentry), vfs_inode + sizeof(struct inode));
+		bpf_probe_read(&lower_inode, sizeof(lower_inode), vfs_inode + sizeof(struct inode) + sizeof(struct dentry *));
+
+		if(!lower_inode && upper_dentry)
```

Comment:
I'd pre-initialize both to NULL so that even if `bpf_probe_read` fails for some reason, we still have a coherent behavior.

Moreover, i'm quite wary about adding the pointer arithmetic above :/ You know, that weird sense of fear with eBPF :D

Reply:
I'd also double check the minimum kernel requirements to see if that holds true on any kernel supported (ie: >= 4.14).